### PR TITLE
crypto: use macro to derive ser/de from ToFromBytes

### PR DESCRIPTION
**Pull request by joyqvq**

### CODE REVIEW

File: fastcrypto/src/serde_helpers.rs
Comment (benr-ml) [inline fastcrypto/src/serde_helpers.rs:173]: pls use SerializationHelper as above
also in the deserializer

Comment (joyqvq) [reply fastcrypto/src/serde_helpers.rs]: came across a limitation when using this in sui. since the length as to be known for SerializationHelper, this does not work with enums where each member has variable length. 

```
pub enum Signature {
    Ed25519SuiSignature,
    Secp256k1SuiSignature,
    Secp256r1SuiSignature,
}
```

`serializer.serialize_bytes(&self.as_bytes())` would be one way to work around this. Ideally I want to use this macro to derive ser/de for enum as well.

**Review by benr-ml — APPROVED** [review]
1. Just a thought, your call - maybe it'd nice to add the length type to trait ToFromBytes (like we have in ToFromByteArray) and use it in the macro instead of adding the length as a parameter.
2. Please check locally that Sui is not trivially broken because of this change before merging.

File: fastcrypto/src/serde_helpers.rs
Comment (benr-ml) [reply fastcrypto/src/serde_helpers.rs]: You mean that the macro won't work for this enum, right? but it will work for each of its types.
That's ok, the goal of the macro is to have a consistent serde for the crypto types, ones that have canonical representations.

Comment (joyqvq) [reply fastcrypto/src/serde_helpers.rs]: :+1 i will have a separate macro customized for sui then. agreed on having serde in fastcrypto to be the most concise one with known length.

**Review by kchalkias — APPROVED** [review]
just resolve conflict re secp

**Review by kchalkias — APPROVED** [review]

**Pull request merged**
